src/wristpy/processing/calibration.py

```diff
@@ -0,0 +1,350 @@
+        devices. If measuring the noise in a bench-top test, this threshold should be
+        about `1.2 * noise`.
+
+        max_iterations (int): The maximum amount of iterations for the closest_point_fit
+        method. Default is 1000, generally should be left at this value.
+
+        error_tolerance (float): Tolerated level of error, when the closest_point_fit
+        method arrives at this value or better, the process ends. Default is 1e-10.
+        Generally should be left at this value.
+
+        min_error (float): Minimum acceptable error. If calibration can not reach this
+        threshold it will error.
+
+
+
+
+    """
+
+    def __init__(
+        self,
+        chunked: bool = False,
+        min_acceleration: float = 0.3,
+        min_calibration_hours: int = 72,
+        min_standard_deviation: float = 0.013,
+        max_iterations: int = 1000,
+        error_tolerance: float = 1e-10,
+        min_error: float = 0.01,
+    ) -> None:
+        """init.
+
+        Attributes:
+            chunked (bool, optional): Whether to use chunked calibration for long
+            recordings.
+
+            min_acceleration (float,optional): Minimum acceleration for sphere criteria.
+
+            min_calibration_hours (int,optional): Minimum hours of data required for
+            calibration.
+
+            min_standard_deviation (float,optional): Minimum standard deviation for
+            no-motion detection.
+
+            max_iterations (int,optional): Maximum number of iterations for
+            optimization.
+
+            error_tolerance (float,optional): Tolerance for optimization convergence.
+
+            min_error (float,optional): Minimum acceptable calibration error.
+
+        Returns:
+            None
+        """
+        self.chunked = chunked
+        self.min_acceleration = min_acceleration
+        self.min_calibration_hours = min_calibration_hours
+        self.min_standard_deviation = min_standard_deviation
+        self.max_iterations = max_iterations
+        self.error_tolerance = error_tolerance
+        self.min_error = min_error
+        self.current_iteration = 0
+
+    def run(self, acceleration: models.Measurement) -> models.Measurement:
+        """Runs acceleration data of the Measurement class based on the settings.
+
+        Chunks the data if chunked = true, otherwise it preforms calibration on the
+        entire dataset. linear transformation is then applied to the data.
+
+        Args:
+            acceleration(Measurement): the accelerometer data containing x,y,z axis
+            data and time stamps.
+
+        Returns:
+            A Measurement object which the original input data, with a linear
+            transformation applied.
+        """
+        data_range = acceleration.time.max() - acceleration.time.min()
+        total_hours = math.ceil(data_range.total_seconds() / 3600)
+
+        if total_hours < self.min_calibration_hours:
+            raise ValueError(
```

Comment:
I think we need "Raises: ..." in the docstring to describe the types of errors that we raise?

Reply:
Yea, added these where necessary